File: examples-sdk/zig/104-pipeline-diamond-topology/main.zig
// 104-pipeline-diamond-topology — Zig SDK equivalent
// Compile: vil compile --from zig --input 104-pipeline-diamond-topology/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var p = vil.Pipeline.init("DiamondSummary", 3095);
    p.sink("summary_sink", 3095, "/diamond");
    p.source("summary_source", "http://localhost:18081/api/v1/credits/ndjson?count=100", "json");
    p.sink("detail_sink", 3096, "/diamond-detail");
    p.source("detail_source", "http://localhost:18081/api/v1/credits/ndjson?count=100", "json");
    p.route("summary_sink.trigger_out", "summary_source.trigger_in", "LoanWrite");
    p.route("summary_source.response_data_out", "summary_sink.response_data_in", "LoanWrite");
    p.route("summary_source.response_ctrl_out", "summary_sink.response_ctrl_in", "Copy");
    p.route("detail_sink.trigger_out", "detail_source.trigger_in", "LoanWrite");
    p.route("detail_source.response_data_out", "detail_sink.response_data_in", "LoanWrite");
    p.route("detail_source.response_ctrl_out", "detail_sink.response_ctrl_in", "Copy");
    p.compile();
}

File: examples-sdk/go/104-pipeline-diamond-topology/main.go
// 104-pipeline-diamond-topology — Go SDK equivalent
// Compile: vil compile --from go --input 104-pipeline-diamond-topology/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	p := vil.NewPipeline("DiamondSummary", 3095)

	p.Sink(vil.SinkOpts{Name: "summary_sink", Port: 3095, Path: "/diamond"})
	p.Source(vil.SourceOpts{Name: "summary_source", URL: "http://localhost:18081/api/v1/credits/ndjson?count=100", Format: "json"})
	p.Sink(vil.SinkOpts{Name: "detail_sink", Port: 3096, Path: "/diamond-detail"})
	p.Source(vil.SourceOpts{Name: "detail_source", URL: "http://localhost:18081/api/v1/credits/ndjson?count=100", Format: "json"})

	p.Route("summary_sink.trigger_out", "summary_source.trigger_in", "LoanWrite")
	p.Route("summary_source.response_data_out", "summary_sink.response_data_in", "LoanWrite")
	p.Route("summary_source.response_ctrl_out", "summary_sink.response_ctrl_in", "Copy")
	p.Route("detail_sink.trigger_out", "detail_source.trigger_in", "LoanWrite")
	p.Route("detail_source.response_data_out", "detail_sink.response_data_in", "LoanWrite")
	p.Route("detail_source.response_ctrl_out", "detail_sink.response_ctrl_in", "Copy")

	p.Compile()
}
